cmd/talosctl/cmd/talos/upgrade.go

```diff
@@ -148,7 +148,12 @@ func upgradeGetActorID(ctx context.Context, c *client.Client, opts []client.Upgr
 func init() {
 	rebootModes := maps.KeysFunc(machine.UpgradeRequest_RebootMode_value, strings.ToLower)
 	sort.Slice(rebootModes, func(i, j int) bool {
-		return machine.UpgradeRequest_RebootMode_value[rebootModes[i]] > machine.UpgradeRequest_RebootMode_value[rebootModes[j]]
+		var (
```

Comment:
I guess this function had a bug. Map entries are uppercase so it was always comparing nil and nil.

Reply:
I have another fix for this issue, which is a bit shorter - just don't do `strings.ToLower` until we finish sorting, and do `stirngs.ToLower` as the last step